src/cpu.cpp

```diff
@@ -1,5 +1,6 @@
 // cpu.cpp
 
+#include "cpu.h"
```

Comment:
Not a linting error, but project structure error. I'm guessing that the lint runner needs a `compile_commands.json` to understand the project structure, this is a similar step I dealt with when building the linting docker image file. The GH action needs to build the project to generate the compile_commands file, otherwise, all header files will trigger this error. I made an adjustment in the yaml, but I didn't have a chance to test it though. @coopeaus Are you able to verify my change works? We might need to force the push and work on the action separately

Reply:
@bgevko pretty sure I know how to fix this, I'm going to add it as a suggestion